src/msgraph_core/tasks/large_file_upload.py

```diff
@@ -0,0 +1,273 @@
+import os
+from typing import Callable, Optional, List, Tuple, Any, Dict
+from io import BytesIO
+from asyncio import Future
+from datetime import datetime, timedelta
+import logging
+
+from kiota_abstractions.serialization.parsable import Parsable
+from kiota_abstractions.method import Method
+from kiota_abstractions.headers_collection import HeadersCollection
+from kiota_abstractions.request_information import RequestInformation
+from kiota_abstractions.serialization.additional_data_holder import AdditionalDataHolder
+
+from kiota_http.httpx_request_adapter import HttpxRequestAdapter as RequestAdapter
+
+from msgraph_core.models import LargeFileUploadSession  # check imports
+
+
+class LargeFileUploadTask:
+
+    def __init__(
+        self,
+        upload_session: LargeFileUploadSession,
+        request_adapter: RequestAdapter,
+        stream: BytesIO,
+        max_chunk_size: int = 409600
+    ):
+        if not isinstance(upload_session, LargeFileUploadSession):
+            raise TypeError("upload_session must be an instance of LargeFileUploadSession")
+        self._upload_session = upload_session
+        self._request_adapter = request_adapter
+        self.stream = stream
+        try:
+            self.file_size = stream.getbuffer().nbytes
+        except AttributeError:
+            self.file_size = os.stat(stream.name).st_size
+        self.max_chunk_size = max_chunk_size
+        cleaned_value = self.check_value_exists(
+            upload_session, 'get_next_expected_range', ['next_expected_range', 'NextExpectedRange']
+        )
+        self.next_range = cleaned_value[0]
+        self._chunks = int((self.file_size / max_chunk_size) + 0.5)
+        self.on_chunk_upload_complete: Optional[Callable[[List[int]], None]] = None
+
+    @property
+    def upload_session(self):
+        return self._upload_session
+
+    @upload_session.setter
+    def upload_session(self, value):
+        self._upload_session = value
+
+    @staticmethod
+    async def create_upload_session(request_adapter: RequestAdapter, request_body, url: str):
```

Comment:
Should we be having this method? I believe the upload session is created using the request builders from the SDK. 

https://learn.microsoft.com/en-us/graph/api/printdocument-createuploadsession?view=graph-rest-1.0&tabs=python